### ISSUE TITLE
An error will be reported if --source='0'

### ISSUE THREAD

**Issue by Concicloud** [ultralytics/yolov5#2875]
## 🐛 Bug
Start-up detect.py an error will be reported if the argument --source ='0'

## Reproduce
Input:
```
python detect.py --source '0' --weights "weights/yolov5s.pt"
```

Output:
```
  File "F:\My Project File\yolov5-master\utils\datasets.py", line 279, in __init__
    if 'youtube.com/' in url or 'youtu.be/' in url:  # if source is YouTube video
TypeError: argument of type 'int' is not iterable
```

## Expected behavior
Turn on the webcam and recognize it in real time.

## Suggestion
I looked at the source code and found that line.
Emmm...
Why not use str()?
So I changed them to this:
```
279    if 'youtube.com/' in str(url) or 'youtu.be/' in str(url):  # if source is YouTube video
```
Then it runs successfully without any errors.
Although it is just a minor mistake, I still wish to change it.

**Comment by glenn-jocher** [comment]
@Concicloud good news 😃! This issue was fixed ✅ a few days ago in PR #2818.  To receive this update you can:
- `git pull` from within your `yolov5/` directory
- `git clone https://github.com/ultralytics/yolov5` again

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other issues you discover or feature requests that come to mind. Happy trainings with YOLOv5 🚀!


@KalenMike @SkalskiP this a knock-on effect of the bug I showed earlier that likely affected about 5000 clones. This was resolved in https://github.com/ultralytics/yolov5/pull/2818, four days ago now, and about 5 days after the bug was introduced (bug report was reported after +4 days, then +1 day for my own delay in seeing the bug report). New users are not aware of fix and continue to file bug reports. 

This is an example of problems that may be improved through stronger CI checks, more attentive issue/discussions reponders and better auto-update methods.

**Comment by nocolour** [comment]
I'm facing the error on windows 10 and ubuntu 20.04 also when use webcam source '0'.
But now tested with ubuntu 20.04, webcam working ready.. Thanks
later will try on windows 10 and raspberry pi 4.
delete the downloaded folder / files. redo the git clone again. It should working now. 
press "Q" to exit sometime not work.. (minor problem) maybe my pc too slow..
